lib/chef/provider/windows_task.rb

```diff
@@ -201,308 +224,303 @@ def action_disable
+        }
+        if new_resource.frequency == :minute
+          trigger_hash[:minutes_interval] = new_resource.frequency_modifier
         end
-      end
 
-      # Comparing execution_time_limit values using Ref: https://github.com/arnau/ISO8601/blob/master/lib/iso8601/duration.rb#L18-L23
-      def execution_time_limit_updated?
-        if new_resource.execution_time_limit.nil?
-          false
-        elsif current_resource.execution_time_limit.nil? && new_resource.execution_time_limit == "PT0S" # when user sets random_dealy to 0 sec
-          false
-        elsif current_resource.execution_time_limit.nil?
-          true
-        else
-          ISO8601::Duration.new(current_resource.execution_time_limit) != ISO8601::Duration.new(new_resource.execution_time_limit)
+        if new_resource.frequency == :hourly
+          minutes = convert_in_minutes(new_resource.frequency_modifier.to_i)
+          trigger_hash[:minutes_interval] = minutes
         end
-      end
 
-      def start_day_updated?
-        current_day = DateTime.strptime(current_resource.start_day, convert_system_date_format_to_ruby_date_format)
-        new_day = parse_day(new_resource.start_day)
-        current_day != new_day
-      end
+        if new_resource.minutes_interval
+          trigger_hash[:minutes_interval] = new_resource.minutes_interval
+        end
 
-      def start_time_updated?
-        time = DateTime.parse(current_resource.start_time).strftime("%H:%M")
-        time != new_resource.start_time
-      end
+        if new_resource.minutes_duration
+          trigger_hash[:minutes_duration] = new_resource.minutes_duration
+        end
 
-      def convert_user_date_to_system_date(date_in_string)
-        parse_day(date_in_string).strftime(convert_system_date_format_to_ruby_long_date)
+        trigger_hash
       end
 
-      def convert_system_date_format_to_ruby_long_date
-        date_format = get_system_short_date_format.dup
-        date_format.sub!("MMM", "%m")
-        common_date_format_conversion(date_format)
-        date_format.sub!("yy", "%Y")
-        date_format
+      def convert_in_minutes(frequency_modifier)
+        frequency_modifier * 60
       end
 
-      def convert_system_date_format_to_ruby_date_format
-        date_format = get_system_short_date_format.dup
-        date_format.sub!("MMM", "%b")
-        common_date_format_conversion(date_format)
-        date_format.sub!("yy", "%y")
-        date_format
-      end
+      #TODO : Try to optimize this method
+      # known issue : Since start_day and time is not mandatory while updating weekly frequency for which start_day is not mentioned by user idempotency
+      # is not gettting maintained as new_resource.start_day is nil and we fetch the day of week from start_day to set and its currently coming as nil and don't match with current_task
+      def task_needs_update?(task)
+        flag = false
+        current_task_trigger = task.trigger(0)
+        new_task_trigger = trigger
```

Comment:
`trigger` is undefined. Pls check if need `current_task_trigger` here

Reply:
trigger is a method  call here